### ISSUE TITLE
Enabling "Sync Playback to Display" causes video stuttering

### ISSUE THREAD

**Issue by chcore** [xbmc/xbmc#19702]
### Describe the bug
Here is a clear and concise description of what the problem is:

As described in the title of this issue, enabling the "Sync Playback to Display" setting results in stuttering video. This happens on the latest x86 LibreELEC nightly as well as Arch (X11 standalone). The setting that changes refresh rate on start/stop is enabled, and my TV supports the resolution/refresh rate of the file being played.

Here's a sample video (note there is nothing special about this video, other than it's a panning shot that makes the issue easy to see): https://www108.zippyshare.com/v/4jjCWRjt/file.html

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Playback should be smooth when this setting is enabled.

## Actual Behavior

There is noticeable stuttering with the setting enabled.

### To Reproduce

1. Enable "Sync Playback to Display" and the setting that changes your refresh rate upon starting/stopping playback
2. Download and play this video (there is nothing special about it, other than it's a panning shot that will make the stuttering easy to see): https://www108.zippyshare.com/v/4jjCWRjt/file.html
3. Observe the stuttering

### Debuglog

The debuglog can be found here:

http://ix.io/3m7x

The above log shows playback of the linked sample file, first with the "Sync Playback to Display" setting enabled, and then with it disabled.

### Screenshots 
Here are some links or screenshots to help explain the problem:

N/A

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

N/A

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: The latest LibreELEC 10 nightly and Arch
 - Kodi version: The latest LibreELEC nightly and 19 (on Arch)

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by lrusak** [comment]
If you have refresh rate switching available why do you need sync playback to display?

**Comment by chcore** [comment]
> If you have refresh rate switching available why do you need sync playback to display?

Occasionally I will run into a video with a frame rate that my display cannot accommodate. Also I have noticed an annoying issue when "Sync Playback to Display" is not enabled, in that I some times get a pause/black frame shortly after switching channels when watching live TV. This is not something I've observed when the syncing was enabled.

I also just want Kodi to be the best it can be :)
